packages/btc/src/base-transfer.ts

```diff
@@ -0,0 +1,22 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+
+type BitcoinTransferRequest = {
+  destinationEvmAddress: string;
+  btcAmount: number;
+};
+
+export function createUriEncodedUtxoRequest(): BaseTransfer {
+  throw new Error("Method not implemented");
+}
+
+export abstract class BaseTransfer {
+  constructor(transfer: BitcoinTransferRequest) {}
+
+  getUriEncodedUtxoRequest(btcTransferRequest: BitcoinTransferRequest): string {
+    throw new Error("Method not implemented");
+  }
+
+  getBTCTransferRequest(): BitcoinTransferRequest {
```

Comment:
ideally there should also be a class called `BitcoinTransfer` or `BitcoinFungibleTransfer`, however, I'd like to know if a generic class like `BaseTransfer` is needed?

Reply:
I was wondering the same once I did the modification. I don't think there is nothing generic about the btc integration that could go into this base class, at least something that I can think now from the top of my head. This is a good point to take into account.

Reply:
@saadjhk I think we do need a base class after all. There is a definition for the btc resource in devnet shared config, which means that upon env definition through env variable, the config class is going to be able to provide the resource information to create either the transfer object or the `bip21` URI for the developer to use. 

I think for this `base` class it could make sense to follow the approach on Anton's pr, where base class basically defines the resource its internal attr. Wdyt?

Reply:
@wainola so the reason why we have a base class in EVM and substrate is due to multiple types of transfers possible, for example fungible, generic therefore an abstraction like `BaseTransfer` can encapsulate the common functionality. However, I'm not sure about bitcoin. I see think we will only have fungible transfers to and from bitcoin network therefore I thought its not needed.

Reply:
From current perspective it looks like we will only support different token transfers on routes to/from BTC. That said, I still think it makes sense to follow same abstraction principles so we can easily extend when/if needed.

Reply:
sounds good then @MakMuftic @wainola Id only suggest friendly/simple naming to make the SDK approachable and accessible